### PR TITLE
[trifid] Add trifid chart

### PR DESCRIPTION
**Pull request by anothertobi**
Signed-off-by: Tobias Nehrlich <tobias.nehrlich@vshn.ch>

#### What this PR does / why we need it:
Adds the trifid chart

#### Checklist

- [x] [DCO](https://github.com/appuio/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR contains starts with chart name e.g. `[chart]`

### CODE REVIEW

File: trifid/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "trifid.fullname" . -}}
+{{- $ingressPath := .Values.ingress.path -}}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+    app: {{ template "trifid.name" . }}
```
Comment (srueg) [inline trifid/templates/ingress.yaml]: Please use the new form of labels: https://helm.sh/docs/topics/chart_best_practices/labels/#standard-labels

File: trifid/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+{{- $endpoint := .Values.trifid.sparql.endpoint -}}
+apiVersion: apps/v1beta2
```
[the remark this answers is not in the document]
Comment (anothertobi) [reply trifid/templates/deployment.yaml]: Changed the `apiVersion`.

File: trifid/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            value: {{ $endpoint.username }}
+            {{- end }}
+          - name: SPARQL_PASSWORD
+            {{- if $endpoint.secretName }}
+            valueFrom:
+              secretKeyRef:
+                name: {{ $endpoint.secretName }}
+                key: password
+            {{- else }}
+            value: {{ $endpoint.password }}
```
[the remark this answers is not in the document]
Comment (anothertobi) [reply trifid/templates/deployment.yaml]: Added a secret and tests for it.

File: trifid/templates/ingress.yaml (hunk 1 above)
Comment (anothertobi) [reply trifid/templates/ingress.yaml]: Changed the labels to the new form (`helm create` uses the old labels...)

File: trifid/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "trifid.fullname" . }})
+  export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+  echo http://$NODE_IP:$NODE_PORT
+{{- else if contains "LoadBalancer" .Values.service.type }}
+     NOTE: It may take a few minutes for the LoadBalancer IP to be available.
+           You can watch the status of by running 'kubectl get svc -w {{ template "trifid.fullname" . }}'
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "trifid.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  echo http://$SERVICE_IP:{{ .Values.service.port }}
+{{- else if contains "ClusterIP" .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "trifid.name" . }},release={{ .Release.Name }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (srueg) [inline trifid/templates/NOTES.txt]: Still references the old labels. IMHO you can remove the NOTES.txt altogether.

File: trifid/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  labels:
+    app.kubernetes.io/name: {{ template "trifid.name" . }}
+    helm.sh/chart: {{ template "trifid.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "trifid.name" . }}
```
Comment (srueg) [inline trifid/templates/deployment.yaml]: Please update the labels here and for the podTemplate as well

File: trifid/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app: {{ template "trifid.name" . }}
```
Comment (srueg) [inline trifid/templates/service.yaml]: Old labels

File: trifid/values.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  tag: 2.3.2
+  pullPolicy: IfNotPresent
+
+trifid:
+  sparql:
+    endpoint:
+      url: https://appuio.ch
+      username: user
+      password: pass
+      # secretName:
```
Comment (srueg) [inline trifid/values.yaml]: I'd propose to rename this to `existingSecret` to be more in line with other Helm charts

File: trifid/values.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+service:
+  type: ClusterIP
+  port: 8080
```
Comment (srueg) [inline trifid/values.yaml]: Let's use port 80 as default since its http

File: trifid/values.yaml (hunk 7 above)
Comment (anothertobi) [reply trifid/values.yaml]: I aligned it with the `tls.secretName`. Should I still change it?

Comment (srueg) [reply trifid/values.yaml]: Ah, I see. Then please add a comment explaining is should reference an existing secret.

File: trifid/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    matchLabels:
+      app.kubernetes.io/name: {{ template "trifid.name" . }}
+      helm.sh/chart: {{ template "trifid.chart" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/managed-by: {{ .Release.Service }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ template "trifid.name" . }}
+        helm.sh/chart: {{ template "trifid.chart" . }}
```
Comment (srueg) [inline trifid/templates/deployment.yaml]: The `"trifid.chart"` contains the version of the chart and would therefore trigger a redeployment on each upgrade (even if unnecessary). Please only add the `/name`, `/instance` and `/managed-by` labels

File: trifid/templates/service.yaml [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app.kubernetes.io/name: {{ template "trifid.name" . }}
```
Comment (srueg) [inline trifid/templates/service.yaml]: Same issue regarding the labels as mentioned before

Comment (anothertobi) [reply trifid/templates/service.yaml]: Added the `helm create` defaults. I was running an outdated helm version... 🤦‍♂

File: trifid/Chart.yaml [hunk 11]
```diff
@@ -1,6 +1,7 @@
-apiVersion: v1
-appVersion: "2.3.2"
-description: Trifid Helm chart
-home: "https://github.com/zazuko/trifid"
+apiVersion: v2
```
Comment (srueg) [inline trifid/Chart.yaml]: `v2` is Helm 3 only apparently: https://helm.sh/docs/topics/charts/#the-apiversion-field

File: trifid/values.yaml [hunk 12]
```diff
@@ -14,6 +7,35 @@ trifid:
...
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: false
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name:
+
+podSecurityContext: {}
```
Comment (srueg) [inline trifid/values.yaml]: Probably makes sense to set a default security context

File: trifid/values.yaml [hunk 13]
```diff
@@ -14,6 +7,35 @@ trifid:
...
+  # fsGroup: 2000
+
+securityContext: {}
+  # capabilities:
```
Comment (srueg) [inline trifid/values.yaml]: Same here, let's make these the default (i.e. uncomment, except `readOnlyRootFilesystem`)

File: trifid/values.yaml [hunk 14]
```diff
@@ -23,21 +45,21 @@ ingress:
...
   tls: []
   #  - secretName: chart-example-tls
   #    hosts:
   #      - chart-example.local
 
 resources: {}
   # limits:
-  #  cpu: 100m
-  #  memory: 128Mi
+  #   cpu: 100m
```
Comment (srueg) [inline trifid/values.yaml]: Let's set some defaults:
```
requests:
  cpu: 500m
  memory: 200Mi
limits:
  cpu: 2000m
  memory: 1Gi
```

**Pull request merged**